net/message/message.go

```diff
@@ -104,12 +104,12 @@ func (m *QueryResponse) Type() byte {
 // SendKeysMessage is sent by both parties to each other to initiate the protocol
 type SendKeysMessage struct {
 	OfferID            types.Hash              `json:"offerID"`
```

Comment:
The `OfferID` above should definitely be tagged as required. The `required` validation will compare a variable to its default initialization value.  In this case, it would compare `OfferID` with an all-zero hash and fail it if it is all zeros. I'll lock down the other fields more in a follow-up PR, because these changes will require non-trivial updates to tests.